src/Adapter/Cache/CacheClearer.php

```diff
@@ -26,47 +26,90 @@
 
 namespace PrestaShop\PrestaShop\Adapter\Cache;
 
-use Tools;
-use Media;
+use PrestaShop\PrestaShop\Core\Cache\Clearer\CacheClearerInterface;
 
 /**
  * Class able to clear application caches.
+ *
+ * @internal
  */
 class CacheClearer
 {
+    /**
+     * @var CacheClearerInterface
+     */
+    private $entireCacheClearer;
+
+    /**
+     * @var CacheClearerInterface
+     */
+    private $symfonyCacheClearer;
+
+    /**
+     * @var CacheClearerInterface
+     */
+    private $mediaCacheClearer;
+
+    /**
+     * @var CacheClearerInterface
+     */
+    private $smartyCacheClearer;
+
+    /**
+     * @param CacheClearerInterface $entireCacheClearer
+     * @param CacheClearerInterface $symfonyCacheClearer
+     * @param CacheClearerInterface $mediaCacheClearer
+     * @param CacheClearerInterface $smartyCacheClearer
+     */
+    public function __construct(
```

Comment:
Watch out, with this you introduce a BC break because of the necessary of having parameters.

Reply:
i know, but do `Adapters` promise BC? i think that modules _should not_ depend on adapters, but if they do, then they _should not_ expect BC as adapters are temporary services that can cahnge/be removed at any time, no? :thinking:

Reply:
There is always a difference between what we think and what users do :D Maybe we can for the moment use a function like `addCache` and call this class `CacheClearerChain`?

Reply:
With this, we sure we don't have BC or anything else

Reply:
i could just leave this `CacheClearer` untouched, because if it is used in some module (even though it shouldnt :D) it will break as you must call `addCacheClearer` to make it work, right?

Reply:
@matks wdyt? im dont really want do add non interfaced method to a service. we could either dispatch event though which you could add more cache clearers or just use composition.